core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/MediaSourceInfoForwarder.java

```diff
@@ -1,21 +1,18 @@
-package com.novoda.noplayer.exoplayer;
+package com.novoda.noplayer.exoplayer.forwarder;
 
-import android.view.Surface;
-
-import com.google.android.exoplayer2.ExoPlaybackException;
 import com.google.android.exoplayer2.Format;
-import com.google.android.exoplayer2.decoder.DecoderCounters;
+import com.google.android.exoplayer2.source.AdaptiveMediaSourceEventListener;
 import com.google.android.exoplayer2.upstream.DataSpec;
 import com.novoda.noplayer.listeners.InfoListeners;
 
 import java.io.IOException;
 import java.util.HashMap;
 
-class InfoForwarder extends ExoPlayerEventForwarder {
+class MediaSourceInfoForwarder implements AdaptiveMediaSourceEventListener {
```

Comment:
Same as others above. We no longer extend from `ExoPlayerEventForwarder` but instead implement the interface we care about directly from `ExoPlayer`.